geode-core/src/main/java/org/apache/geode/internal/cache/TombstoneService.java

```diff
@@ -803,6 +856,11 @@ public void unscheduleTombstones(final LocalRegion r) {
       });
     }
 
+    public void setTOmbstoneTimeout(long expiryTime) {
+      this.EXPIRY_TIME = expiryTime;
+      this.PURGE_INTERVAL = Math.min(DEFUNCT_TOMBSTONE_SCAN_INTERVAL, expiryTime);
```

Comment:
If you keep this, allowing these variables to be changed after the constructor, then change the constructor to calll this method instead of duplicating the code that computes PURGE_INTERVAL.